app/src/main/java/com/vultisig/wallet/data/repositories/AccountsRepository.kt

```diff
@@ -30,30 +31,67 @@ internal interface AccountsRepository {
 internal class AccountsRepositoryImpl @Inject constructor(
     private val vaultDb: VaultDB,
     private val balanceRepository: BalanceRepository,
-    private val chainAccountAddressRepository: ChainAccountAddressRepository,
     private val tokenPriceRepository: TokenPriceRepository,
     private val coinToChainAccountMapper: CoinToChainAccountMapper,
 ) : AccountsRepository {
 
+    private val defaultChains =
+        listOf(Chain.thorChain, Chain.bitcoin, Chain.bscChain, Chain.ethereum, Chain.solana)
+
     private fun getVault(vaultId: String): Vault =
         checkNotNull(vaultDb.select(vaultId)) {
             "No vault with id $vaultId"
         }
 
-    override fun loadAccounts(vaultId: String): Flow<List<Account>> = flow {
-        val vault = getVault(vaultId)
+    private fun setDefaultChains(vault: Vault): Vault {
+        if (vault.coins.isNotEmpty()) {
+            return vault
+        }
+        var updateVault = vault
+        defaultChains.forEach { chain ->
+            when (chain) {
+                Chain.thorChain -> {
+                    THORCHainHelper(vault.pubKeyECDSA, vault.hexChainCode).getCoin()?.let { coin ->
+                        updateVault = updateVault.copy(coins = updateVault.coins + coin)
+                    }
+                }
+
+                Chain.bitcoin -> {
+                    utxoHelper.getHelper(vault, CoinType.BITCOIN).getCoin()?.let { coin ->
+                        updateVault = updateVault.copy(coins = updateVault.coins + coin)
+                    }
+                }
+
+                Chain.bscChain -> {
+                    // TODO: add it
+                }
+
+                Chain.ethereum -> {
+                    // TODO: add it
+                }
+
+                Chain.solana -> {
+                    // TODO: add it
+                }
+
+                else -> {
+                    //do nothing
+                }
+            }
+        }
+        vaultDb.upsert(updateVault)
+        return updateVault
+    }
 
+    override fun loadAccounts(vaultId: String): Flow<List<Account>> = flow {
+        var vault = getVault(vaultId)
+        vault = setDefaultChains(vault)
```

Comment:
`setDefaultChains` will only apply the logic when `vault.coins` is empty , which effectively means it will only add default when the vault just created.  Or when user remove all the coins from the vault. 

I think it is ok to stay here ,  or I can move it to `GenertingKeyViewModel` ,when we save the vault for the first time